refactor(napi/oxlint): remove temp var

overlookmotel opened the pull request:
Remove `processedBuffer` variable, which was only present to satisfy type-checker, and is unnecessary at runtime.

overlookmotel commented:
* **#12590** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12590?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12590?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=12590).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 commented on napi/oxlint2/src-js/index.ts:
```diff
@@ -189,25 +189,22 @@ const diagnostics: DiagnosticReport[] = [];
 
 // Run rules on a file.
-function lintFile(filePath: string, bufferId: number, buffer: Uint8Array | undefined | null, ruleIds: number[]) {
+function lintFile(filePath: string, bufferId: number, buffer: Uint8Array | null, ruleIds: number[]) {
```
let me ask Brooklyn about this change (removing `undefined`)

camc314 reviewed (commented):
Hmm i'm actually not a massive fan of this change, as it feels like we're lying to typescript in a bunch more places than previously.

Before (yes we had the extra var), but I think it was slightly clearer about what was what type where.

What was the perf diff of this change?

camc314 commented:
### Merge activity

* **Jul 30, 10:45 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(napi/oxlint): remove temp var (#12590)

Remove `processedBuffer` variable, which was only present to satisfy type-checker, and is unnecessary at runtime.

Note: NAPI-RS translates `Option::None` on Rust side to `null` in JS. I'm not sure why the typing NAPI-RS produces for `buffer` argument is `Uint8Array | undefined | null`. It's never `undefined`, so `Uint8Array | null` is more accurate.
